### PR TITLE
Permission management

### PR DESCRIPTION
**Pull request by axel3rd**
Proposal for [Capacity to manage (edit/remove) only by admin](https://github.com/Linfar/TeamCity.SonarQubePlugin/issues/14) *bug*.

Granted on **Permission.EDIT_PROJECT**, so give the capacity to create a dedicated role.

The best could be to create/register a new specific permission, but not sure possible with TC 8.1 (update :   [not possible](https://devnet.jetbrains.com/message/5549677;jsessionid=88C61A415D3ED86421527648A6BBC731#5549677))

### CODE REVIEW

**Comment by Linfar** [comment]
Hi, 
Thank you for contributing and please excuse me for long response, I will comment changes inline

File: sonar-plugin-server/src/main/java/jetbrains/buildserver/sonarplugin/sqrunner/ManageSQSActionController.java [hunk 1]
```diff
@@ -70,14 +79,17 @@ private static String getAction(final @NotNull HttpServletRequest request) {
...
-        if (ajaxResponse == null) {
+        final SProject project = getProject(request);
+        if (ajaxResponse == null || project == null) {
             return;
         }
 
-        final SProject project = getProject(request);
-        if (project == null) {
+        // Security test (user without management permission could access this controller)
+        if (!AuthUtil.hasGlobalPermission(securityContext.getAuthorityHolder(), Permission.CHANGE_SERVER_SETTINGS)){
```
Comment (Linfar) [inline sonar-plugin-server/src/main/java/jetbrains/buildserver/sonarplugin/sqrunner/ManageSQSActionController.java]: `CHANGE_SERVER_SETTINGS` usually means global server admin which has too many rights. As SonarQube Server is project-level configuration `EDIT_PROJECT` is much more suitable permission for this issue. `AuthUtil.hasPermissionToManageProject(securityContext.getAuthorityHolder(), project)` could be used.

File: sonar-plugin-server/src/main/java/jetbrains/buildserver/sonarplugin/sqrunner/ServerManagementProjectTab.java [hunk 2]
```diff
@@ -77,4 +86,8 @@ public void fillModel(@NotNull final Map<String, Object> model, @NotNull final H
         }
         return infoMap;
     }
+
+    private boolean userHasPermission() {
+        return AuthUtil.hasGlobalPermission(securityContext.getAuthorityHolder(), Permission.CHANGE_SERVER_SETTINGS);
```
Comment (Linfar) [inline sonar-plugin-server/src/main/java/jetbrains/buildserver/sonarplugin/sqrunner/ServerManagementProjectTab.java]: Same as in ManageSQSActionController:88 - `EDIT_PROJECT` permission should be used

**Comment by Linfar** [comment]
@axel3rd I've merged your branch with 2 minor fixes: 
Internal ProjectID is expected in AuthUtil
Edit actions should not be visible for upper projects without edit permission (eg a project admin has edit permission for a child project only while server is defined in a parent project)

**Comment by axel3rd** [comment]
@Linfar : Thanks for fix (I'm very newbie in TeamCIty plugin dev ...). 

Could the build [#23](https://teamcity.jetbrains.com/viewLog.html?buildId=541375&tab=buildResultsDiv&buildTypeId=TeamCityPluginsByJetBrains_TeamCitySonarQubePlugin_Build) _pinned_ ? The download link on home/*readme.txt* (*.lastPinned*) is build #20 => The documentation content about permission is not contained in this build).

**Comment by Linfar** [comment]
@axel3rd done

**Comment by axel3rd** [comment]
@Linfar Thanks

**Pull request merged**